etcdmain/etcd.go

```diff
@@ -379,3 +387,11 @@ func genClusterString(name string, urls types.URLs) string {
 	}
 	return strings.Join(addrs, ",")
 }
+
+// handleInterrupt calls the handler on receiving a SIGTERM.
+func handleInterrupt(h func()) {
+	notifier := make(chan os.Signal)
+	signal.Notify(notifier, os.Interrupt, syscall.SIGTERM)
+	<-notifier
+	h()
```

Comment:
After stopping etcd, you need to kill yourself with the same signal that triggered the handler. (http://www.cons.org/cracauer/sigint.html)